scripts/yg-privacyenhanced.py

```diff
@@ -29,21 +29,54 @@ def select_input_mixdepth(self, available, offer, amount):
         the first mixdepth available after the largest such interval.
         This forces the biggest UTXOs to stay in a bulk of few mixdepths so
         that the maker can always maximize the size of his orders even when
-        some coins are sent from the last to the first mixdepth"""
+        some coins are sent from the last to the first mixdepth.
+        In the edge case where several mixdepths are as efficient, we
+        select the one which maximize the balance of the receiving mixdepth,
+        except when all mixdepths can be used, in which case we estimate which
+        one was less used to received orders for the longest and we use it to
+        use the otherwise locked amount and consolidate UTXOs."""
+        # Total number of mixdepths
+        mix_total = self.wallet_service.mixdepth + 1
+        # Balances in each mixdepth
+        mix_balance = self.get_available_mixdepths()
         # We sort the available depths for linear scaling of the interval search
         available = sorted(available.keys())
+        # Case where all mixdepths can fill the order: we search the mixdepth which
+        # stopped receiving funds the longest by looking at balances around the
```

Comment:
I don't get "which stopped receiving funds the longest" part of this comment. It just looks at mixdepth total amount, not when funds were received.